lib/src/generate/watch_impl.dart

```diff
@@ -195,17 +195,45 @@ class WatchImpl {
         _logger.fine('Setting up watcher at $absolutePackagePath');
 
         // Ignore all subfolders which are other packages.
-        var pathsToIgnore = absolutePackagePaths.values.where((path) =>
-            path != absolutePackagePath &&
-            path.startsWith(absolutePackagePath));
+        var pathsToIgnore = absolutePackagePaths.values
+            .where((path) =>
+                path != absolutePackagePath &&
+                path.startsWith(absolutePackagePath))
+            .toList();
 
         var watcher = _directoryWatcherFactory(absolutePackagePath);
         watchers.add(watcher);
         _allListeners.add(watcher.events.listen((WatchEvent e) {
+          var changePath = e.path;
+
+          // Convert `packages` paths to absolute paths.
+          () {
```

Comment:
Oh I just used that to encapsulate all the local variables so I didn't have to worry about using somewhat generic names. Really though I should have just broken this out into a helper method or something, I will do that instead.